firefox-ios/Client/Telemetry/TelemetryWrapper.swift

```diff
@@ -100,10 +100,25 @@ class TelemetryWrapper: TelemetryWrapperProtocol, FeatureFlaggable {
         GleanMetrics.Search.defaultEngine.set(defaultEngine?.engineID ?? "unavailable")
 
         // Get the legacy telemetry ID and record it in Glean for the deletion-request ping
-        if let uuidString = UserDefaults.standard.string(forKey: "telemetry-key-prefix-clientId"), let uuid = UUID(uuidString: uuidString) {
+        if let uuidString = UserDefaults.standard.string(forKey: "telemetry-key-prefix-clientId"),
+           let uuid = UUID(uuidString: uuidString) {
             GleanMetrics.LegacyIds.clientId.set(uuid)
         }
 
+        // Set or generate profile id used for usage reporting
+        if profile.prefs.boolForKey(AppConstants.prefSendUsageData) ?? true {
```

Comment:
this is passed in as `sendUsageData`:

```suggestion
        if sendUsageData {
```